lib/buffer_update_component.coffee

```diff
@@ -5,11 +6,14 @@ module.exports =
 class BufferUpdateComponent
   constructor: (@editor, @analysisAPI) ->
     @events = new CompositeDisposable()
-    @events.add @editor.onDidStopChanging =>
+    updateFile = =>
       descriptors = @editor.getRootScopeDescriptor()
       # We only support pure Dart files for now
       return unless Utils.isCompatible(editor)
       @analysisAPI.updateFile @editor.getPath(), @editor.getText()
 
+    @events.add @editor.onDidStopChanging _.debounce(updateFile, 250,
+      maxWait: 1000)
```

Comment:
What I'm trying to avoid with a long max timeout is when the tool tells me the code is broken, _while_ I'm in the middle of typing it. It'll underline a bunch of stuff with red even though I know the code doesn't compile yet. Very short timeouts can feel to me as if the tool is penalizing me for typing too slowly.

That said, I'm not sure what the best time is; we can certainly determine a reasonable one by trial and error - committing a value and iterating.